From 91430dfb2362a6778a9b9a816f1ca58f66ee78cd Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Mon, 24 Aug 2026 16:44:21 +0200
Subject: [PATCH 38/38] net: sfp-bus: handle extended compliance codes for
 100/25G acc cables

Handle extended compliance codes 0x08 and 0x19 in sfp_parse_support, to
support 100G/25G active copper cables.

This was tested with FS QSFP-100G-4SAC03 100G QSFP to 4x SFP28 adapter
cable, which ships programmed with ecc 0x19.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/net/phy/sfp-bus.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/phy/sfp-bus.c b/drivers/net/phy/sfp-bus.c
index 79bc481de99b..a4d8157b4260 100644
--- a/drivers/net/phy/sfp-bus.c
+++ b/drivers/net/phy/sfp-bus.c
@@ -273,6 +273,11 @@ void sfp_parse_support(struct sfp_bus *bus, const struct sfp_eeprom_id *id,
 	case SFF8024_ECC_100GBASE_ER4_25GBASE_ER:
 		phylink_set(modes, 100000baseLR4_ER4_Full);
 		break;
+	case SFF8024_ECC_100G_25GAUI_C2M_ACC_BER_5E_5:
+	case SFF8024_ECC_100G_25GAUI_C2M_ACC_BER_1E_12:
+		phylink_set(modes, 25000baseCR_Full);
+		__set_bit(PHY_INTERFACE_MODE_25GBASER, interfaces);
+		break;
 	case SFF8024_ECC_100GBASE_CR4:
 		phylink_set(modes, 100000baseCR4_Full);
 		fallthrough;
-- 
2.51.0

